superset-frontend/packages/superset-ui-core/src/components/SafeMarkdown.tsx

```diff
@@ -16,38 +16,44 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
-import React from 'react';
-import ReactMarkdown, { MarkdownAbstractSyntaxTree } from 'react-markdown';
-// @ts-ignore no types available
-import htmlParser from 'react-markdown/plugins/html-parser';
-
+import React, { useMemo } from 'react';
+import ReactMarkdown from 'react-markdown';
+import rehypeSanitize, { defaultSchema } from 'rehype-sanitize';
+import rehypeRaw from 'rehype-raw';
+import { merge } from 'lodash';
 import { FeatureFlag, isFeatureEnabled } from '../utils';
 
 interface SafeMarkdownProps {
   source: string;
+  htmlSanitization?: boolean;
+  htmlSchemaOverrides?: typeof defaultSchema;
 }
 
-function isSafeMarkup(node: MarkdownAbstractSyntaxTree) {
-  return node.type === 'html' && node.value
-    ? !/(href|src)="(javascript|vbscript|file):.*"/gim.test(node.value)
-    : true;
-}
+function SafeMarkdown({
+  source,
+  htmlSanitization = true,
+  htmlSchemaOverrides = {},
+}: SafeMarkdownProps) {
```

Comment:
It'd be nice to have RTL tests that test something along the following lines:
- the default use case where source is rendering correctly
- sanitization is disabled -> how do things change
- a schema override before and after

Reply:
I'll work on this in a follow-up 👍🏼

Reply:
@villebro `superset-ui/core` only guarantee 100% coverage on the `js/ts` file.

Reply:
> @villebro `superset-ui/core` only guarantee 100% coverage on the `js/ts` file.

Oh right! Thanks for following up @zhaoyongjie ! 👍